src/SqlPersistence/ScriptRunner.cs

```diff
@@ -29,30 +40,37 @@ public static class ScriptRunner
             Guard.AgainstNullAndEmpty(nameof(scriptDirectory), scriptDirectory);
             sqlDialect.ValidateTablePrefix(tablePrefix);
 
-            using (var connection = await connectionBuilder.OpenConnection().ConfigureAwait(false))
-            using (var transaction = connection.BeginTransaction())
+            if (shouldInstallOutbox)
             {
-                if (shouldInstallOutbox)
-                {
-                    await InstallOutbox(scriptDirectory, connection, transaction, tablePrefix, sqlDialect).ConfigureAwait(false);
-                }
+                await Execute<StorageType.Outbox>(InstallOutbox, scriptDirectory, tablePrefix, sqlDialect, connectionBuilder).ConfigureAwait(false);
```

Comment:
I know this is used to avoid repetition, but I'd prefer to have that "execute" method inlined. It's really hard to follow this code :/

Or, the minimum would be to give this method a more descriptive name like `ExecuteScriptTransactionally` or something like that.

Reply:
ping

Reply:
I am going with `ExecuteInSeparateConnection`.